fulfillment-service/it/it_idp_login_helpers.go

```
@@ -0,0 +1,459 @@
+		if code == http.StatusOK {
+			return nil
+		}
+		return fmt.Errorf("IdP %q not yet in Keycloak (status %d)", alias, code)
+	}, backoff.WithContext(bo, ctx))
+}
+
+// SimulateOIDCLogin drives the OIDC authorization code flow programmatically to obtain a
+// Keycloak JWT for an external IdP user. The redirect chain is:
+//
+//  1. Test runner  → GET KC auth endpoint (kc_idp_hint=<alias>)
+//  2.              ← 302 to mockoidc /oidc/authorize
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gofmt)